### PR TITLE
FIX: Make email_valid handling consistent

### PR DESCRIPTION
**Pull request by davidtaylorhq**
Previously we were checking truthiness in some places, and `== true` in
others. That can lead to some inconsistent UX where the interface says
the email is valid, but account creation fails.

This commit ensures values are boolean when set, and raises an error for
other value types.

If this safety check is triggered, it means the specific auth provider
needs to be updated to pass booleans.

### CODE REVIEW

File: lib/auth/result.rb
Comment (CvX) [inline lib/auth/result.rb:59]: Is `"false"` the only invalid input this can receive? If not, perhaps allowing a certain set of values and raising on anything else would be better? 🤷

Comment (davidtaylorhq) [reply lib/auth/result.rb:59]: I made this PR because someone had configured the `discourse-oauth2-basic` plugin to use use the actual email address as the `email_verified` value.

Ideally, only `true` and `false` would be valid values. My worry is that enforcing that will break some auth plugins... maybe we should just rip off the band-aid and identify all the issues.

Comment (CvX) [reply lib/auth/result.rb:59]: I'd either go all in and let any invalid auth plugins burn 😉 or accept all input and try to auto-correct (i.e. instead of raising on `"false"` treat it as `false`)

One is more "correct" but provides you with with potential work for the years to come 😜

Comment (davidtaylorhq) [reply lib/auth/result.rb:59]: Let's just go all-in. Auth, and email validation, are security-critical things so I think it's better for things to fail than have weird edge cases. Will wait until after the holidays to merge though - we don't want to accidentally make extra work for ourselves over the next few days.

**Comment by davidtaylorhq** [comment]
To be merged after the holidays, just in case it causes any auth plugin issues. Marking draft to prevent accidental merging

**Review by CvX — APPROVED** [review]

**Pull request merged**
